### PR TITLE
Registry refactor hook up

### PR DESCRIPTION
**Pull request by garethfuller**
# Description

Hooking up the new token related composables throughout the app. Apologies for this ridiculously large PR, it was difficult to do it in smaller chunks.

When reviewing, because there are so many changes, it's best if you just look at files / areas of the app you are familiar with. Also, below is a list of the key files that make up the new tokens system.

In general, it's best if you review by testing as much of the app functionality as possible and then reviewing files when you find bugs.

#### Queries
- [src/composables/queries/useTokenListsQuery.ts](https://github.com/balancer-labs/frontend-v2/pull/570/files#diff-2e6f83a442e5d39f64d35726319c334f307980461b36f4c9a9bde96869471523R1)
- [src/composables/queries/useTokenPricesQuery.ts](https://github.com/balancer-labs/frontend-v2/pull/570/files#diff-78741e101253bf792ba750439a46b2d3f49d837a9483d1c31aeb46adc458e91dR2)
- [src/composables/queries/useBalancesQuery.ts](https://github.com/balancer-labs/frontend-v2/pull/570/files#diff-59e35f056bcefd366bc870948a2532e9ca5f21029a4df6b8ff9be0691a64fe67R1)
- [src/composables/queries/useAllowancesQuery.ts](https://github.com/balancer-labs/frontend-v2/pull/570/files#diff-bfe6ba93eb7c8c7f1620f2fa8a6d14ecb551532253e08b6be437c644dd934aa0R1)

#### Providers
- [src/providers/token-lists.provider.ts](https://github.com/balancer-labs/frontend-v2/pull/570/files#diff-afce42ee6d2acb664573385562a627e27168da139a374ceacc03377128dabb2bR1)
- [src/providers/tokens.provider.ts](https://github.com/balancer-labs/frontend-v2/pull/570/files#diff-b2467c63b8599895fd8bd59da8db65589337abe714425b6ccea5254744dad72dR1)

#### Other important changes
- [src/services/token/concerns/allowances.concern.ts](https://github.com/balancer-labs/frontend-v2/pull/570/files#diff-02e058a0a8bc19ac43af397c4b04e2a27429f05bc5bc80f9ac64a7dde02cc969R1)
- [src/services/token/concerns/balances.concern.ts](https://github.com/balancer-labs/frontend-v2/pull/570/files#diff-35484367783db333552d4b68523a0b54fc9e6177b9b534195717d1f65dc88abaR1)
- [src/components/modals/SelectTokenModal/SelectTokenModal.vue](https://github.com/balancer-labs/frontend-v2/pull/570/files#diff-4636162f18fb623a319982cc9c114af2ede0e85daeeddeee34a9fb33d8c22a9cR26)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

## How should this be tested?

- [ ] Generally test all core functionality is working as expected. In particular all different types of trades (incl. Gnosis), investing/withdrawing and token search functionality.
- [ ] Test injecting a token via address in token search modal
- [ ] Test trade interface deep links

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [x] My changes generate no new console warnings
- [x] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/composables/queries/useAllowancesQuery.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  /**
+   * QUERY INPUTS
+   */
+  const queryKey = reactive(
+    QUERY_KEYS.Account.Allowances(account, contractAddesses, tokenAddresses)
+  );
+
+  const queryFn = async () => {
+    console.log('Fetching', tokenAddresses.value.length, 'allownances');
```
Comment (evgenyboxer) [inline src/composables/queries/useAllowancesQuery.ts:44]:
```suggestion
    console.log('Fetching', tokenAddresses.value.length, 'allowances');
```

File: src/composables/useUrls.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { configService } from '@/services/config/config.service';
+
+export default function useUrls() {
+  function resolve(uri: string): string {
+    if (!uri) return '';
+    return uri
+      .replace('ipfs://', `https://${configService.env.IPFS_NODE}/ipfs/`)
+      .replace('ipns://', `https://${configService.env.IPFS_NODE}/ipns/`);
```
Comment (evgenyboxer) [inline src/composables/useUrls.ts:8]: I think in this instance `resolve` can live outside the composable right? just to reduce the number of times we redeclare this function.

File: src/components/forms/pool_actions/WithdrawForm.vue [hunk 3]
```diff
@@ -59,7 +59,10 @@
               <span class="break-words">
                 {{ amountUSD(i) === 0 ? '-' : fNum(amountUSD(i), 'usd') }}
               </span>
-              <span class="text-xs text-gray-400">
+              <span
+                v-if="pool.poolType !== 'Stable'"
```
Comment (evgenyboxer) [inline src/components/forms/pool_actions/WithdrawForm.vue:63]: Would be great to have this as an enum/constant

File: src/components/modals/TradePreviewModalGP.vue [hunk 4]
```diff
@@ -148,17 +148,17 @@
       </BalCard>
       <div v-if="trading.requiresApproval.value" class="flex justify-between">
         <BalBtn
-          v-if="!isUnlocked || approved"
+          v-if="!isApproved || approved"
```
Comment (evgenyboxer) [inline src/components/modals/TradePreviewModalGP.vue:151]: I guess `isApproved` is better, I just wanted to differentiate between "I have just approved" (`approved`) and "it was previously approved / just approved it" (`isApproved`)

Maybe what's confusing is `approved` and should be renamed to `recentlyApproved` or `justApproved`

File: src/composables/queries/useTokenListsQuery.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  const queryFn = async () => {
+    console.log('Fetching tokenLists');
+    return await tokenListService.getAll();
+  };
+
+  const queryOptions = reactive({
+    enabled: true,
+    refetchOnMount: false,
+    refetchOnWindowFocus: false,
+    refetchOnReconnect: false,
```
Comment (evgenyboxer) [inline src/composables/queries/useTokenListsQuery.ts:30]: Based on this artice - https://medium.com/in-the-weeds/fetch-a-query-only-once-until-page-refresh-using-react-query-a333d00b86ff

We might also need to set: `staleTime` to something like "24h".

Might also worth exporting this into something like `FETCH_ONCE_OPTIONS`, etc.

**Review by evgenyboxer — APPROVED** [review]
lgtm! awesome work @garethfuller 

I've gone through the code and sanity checked the app... have not found any issues yet.

File: src/composables/useUrls.ts (hunk 2 above)
Comment (garethfuller) [reply src/composables/useUrls.ts:8]: Yup good point.

File: src/components/forms/pool_actions/WithdrawForm.vue (hunk 3 above)
Comment (garethfuller) [reply src/components/forms/pool_actions/WithdrawForm.vue]: Agreed: https://github.com/balancer-labs/frontend-v2/pull/570/commits/fde9bfe897adbb9a83796085b0db32c1eaf8717b

File: src/components/modals/TradePreviewModalGP.vue (hunk 4 above)
Comment (1saf) [reply src/components/modals/TradePreviewModalGP.vue:151]: agree

Comment (garethfuller) [reply src/components/modals/TradePreviewModalGP.vue]: Yeah agree, instead I've just reverted back to `isUnlocked` we can change later if it makes sense. https://github.com/balancer-labs/frontend-v2/pull/570/commits/0bc070a09b2a3ff1aa74d638bebae8fa688e76ce

**Review by 1saf — APPROVED** [review]

File: src/composables/queries/useTokenListsQuery.ts (hunk 5 above)
Comment (garethfuller) [reply src/composables/queries/useTokenListsQuery.ts]: Done: https://github.com/balancer-labs/frontend-v2/pull/570/commits/8fef34400f5508c8fb79726cc9016e6e10cfa2a4

**Pull request merged**